### PR TITLE
POLIO-1494 Campaigns yearly calendar

### PR DESCRIPTION
**Pull request by beygorghor**
For now calendar is showing 4 month, we should be able to select other range like semester, quarter, ...

Related JIRA tickets : POLIO-1494

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- adapt number of columns to display yearly and semester
- use toggle buttons to change period type
- deep link period type
- relay-out calendar table to fit a whole year on the page

## How to test

Got to calendar page and play with periods, also test embedded calendar 

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/49a634c9-626e-44d4-ae93-23d4a605bda3

### CODE REVIEW

File: .eslintrc.json
Comment (beygorghor) [inline .eslintrc.json:1]: this is not part of this PR but makes sense

File: plugins/polio/js/src/domains/Calendar/campaignCalendar/Head.tsx [hunk 1]
```diff
@@ -95,7 +98,11 @@ export const Head: FunctionComponent<Props> = ({
             <TableRow
                 className={classnames(classes.tableRow, classes.tableRowSmall)}
             >
-                <HeadStaticFieldsCells orders={orders} isPdf={isPdf} />
+                <HeadStaticFieldsCells
+                    orders={orders}
+                    isPdf={isPdf}
+                    router={router}
```
Comment (quang-le) [inline plugins/polio/js/src/domains/Calendar/campaignCalendar/Head.tsx:104]: I don't get this: `router` is added to the props, but the changes to `HeadStaticFields` don't reflect this.

File: plugins/polio/js/src/domains/Calendar/campaignCalendar/cells/HeadStaticFields.tsx
Comment (quang-le) [inline plugins/polio/js/src/domains/Calendar/campaignCalendar/cells/HeadStaticFields.tsx:1]: so line 67, 68 expected `router` prop but didn't receive it?

**Review by quang-le — APPROVED** [review]
Some minor visual bugs that can be handled separately:
Screenshot 2024-04-11 at 12 00 09
Screenshot 2024-04-11 at 11 59 16
